[WIP] - Mention dependency resolution for new distributions (#7046)
ballerina-platform/ballerina-dev-website: swan-lake/learn-the-platform/source-code-dependencies/manage-dependencies.md

version = "1.5.0”
```

> **Building with a new distribution version**

Reviewer: Wrote the following with the help of ChatGPT 

### Upgrading packages to use new Swan Lake Update distributions

When a new Swan Lake Update distribution is released, it may include incompatible language improvements. When upgrading an existing package to use new Swan Lake Update distributions, the Ballerina compiler automatically updates the `Dependencies.toml` file with the latest compatible minor versions of dependencies for the new distribution. This enables existing projects to benefit from improvements in the new Update without significant modifications.

> ### Upgrading packages to use new Swan Lake Update distributions

Reviewer 2: ```suggestion
### Upgrading packages to use new Swan Lake update distributions
```

> When a new Swan Lake Update distribution is released, it may include incompatible language improvements. When upgrading an existing package to use new Swan Lake Update distributions, the Ballerina compiler automatically updates the Dependencies.toml file with the latest compatible minor versions of dependencies for the new distribution. This enables existing projects to benefit from improvements in the new Update without significant modifications.

Reviewer 2: ```suggestion
When a new Swan Lake Update distribution is released, it may include incompatible language improvements. When upgrading an existing package to use new Swan Lake Update distributions, the Ballerina compiler automatically updates the `Dependencies.toml` file with the latest compatible minor versions of the dependencies of the new distribution. This enables the existing projects to benefit from improvements in the new update releases without requiring significant modifications.
```

> When a new Swan Lake Update distribution is released, it may include incompatible language improvements. When upgrading an existing package to use new Swan Lake Update distributions, the Ballerina compiler automatically updates the `Dependencies.toml` file with the latest compatible minor versions of the dependencies of the new distribution. This enables the existing projects to benefit from improvements in the new update releases without requiring significant modifications.

Reviewer 2: ```suggestion
When a new Swan Lake update distribution is released, it may include incompatible language improvements. When upgrading an existing package to use new Swan Lake update distributions, the Ballerina compiler automatically updates the `Dependencies.toml` file with the latest compatible minor versions of the dependencies of the new distribution. This enables the existing projects to benefit from improvements in the new update releases without requiring significant modifications.
```
